nengo/networks/assoc_mem.py

```diff
@@ -94,49 +106,139 @@ def __init__(self, input_vectors, output_vectors=None,  # noqa: C901
-            self.elem_input = nengo.Node(
-                size_in=self.n_items, label="element input")
-            self.elem_output = nengo.Node(
-                size_in=self.n_items, label="element output")
-            self.utilities = self.elem_output
+            self.ens_input = nengo.Node(
+                size_in=self.n_items, label="ensembles input")
+            self.ens_utilities = nengo.Node(
+                size_in=self.n_items, label="ensembles utilities")
 
             self.am_ensembles = []
             label_prefix = "" if label is None else label + "_"
-            filt_scale = 15
-            filt_step_func = lambda x: filtered_step(x, 0.0, scale=filt_scale)
+
             for i in range(self.n_items):
                 e = nengo.Ensemble(n_neurons, 1, label=label_prefix + str(i))
                 self.am_ensembles.append(e)
 
                 # Connect input and output nodes
                 nengo.Connection(self.bias_node, e, transform=-threshold[i])
-                nengo.Connection(self.elem_input[i], e)
-                nengo.Connection(
-                    e, self.elem_output[i], function=filt_step_func)
+                nengo.Connection(self.ens_input[i], e, synapse=None)
+                nengo.Connection(e, self.ens_utilities[i], synapse=None)
 
             if inhibitable:
                 # Input node for inhibitory gating signal (if enabled)
                 self.inhibit = nengo.Node(size_in=1, label="inhibit")
-                nengo.Connection(self.inhibit, self.elem_input,
+                nengo.Connection(self.inhibit, self.ens_input,
                                  transform=-np.ones((self.n_items, 1))
-                                 * self._inhib_scale)
+                                 * self._inhib_scale, synapse=None)
                 # Note: We can use a decoded connection here because all the
                 # am_ensembles have [1] encoders
             else:
                 self.inhibit = None
         self.add_input_mapping("input", input_vectors, input_scales)
         self.add_output_mapping("output", output_vectors)
 
+    @staticmethod
+    def linear_func(x_shift=0.0, x_scale=1.0):
+        """Returns a linear mapping function.
+
+        Returns a linear mapping function for ease of use when creating output
+        mappings. The returned function takes in a non-optional 'x' parameter,
+        as well as optional parameters listed below.
+
+        Parameters
+        ----------
+        x_shift: float, optional
+            Amount to shift the linear map on the 'x'-axis.
+            E.g. If the value of x_shift = 0.3, then the output of the
+                 mapping function would be 0 at x == 0.3.
+        x_scale: float, optional
+            Scaling factor to be applied to the 'x' input. Affects the slope
+            of the linear mapping function.
+        """
+        return lambda x, x_shift=x_shift, x_scale=x_scale: \
+            (x_scale * (x - x_shift))
```

Comment:
I'm pretty sure you're right that it's not needed here, and I'd say we should only use this idiom when it is required (otherwise it looks odder than it needs to...)